src/main/java/by/tms/insta/service/UserService.java

```diff
@@ -0,0 +1,47 @@
+package by.tms.insta.service;
+
+import by.tms.insta.dao.UserStorage;
+import by.tms.insta.entity.User;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.List;
+
+public class UserService {
+    private final UserStorage userStorage;
+    private final UserValidator userValidator;
+
+    public UserService(@Qualifier("hibernateUserStorage") UserStorage userStorage, UserValidator userValidator) {
+        this.userStorage = userStorage;
+        this.userValidator = userValidator;
+    }
+
+    public boolean save(User user) {
+        if (!userValidator.loginAndNameValidate(user.login, user.name) || !userValidator.passwordValidate(user.password)) {
+            return false;
+        }
+        if (!userStorage.userExists(user)) {
```

Comment:
It looks like if user exists: method is gonna return false and doesn't save. You must remove negation